src/signalrclient/default_websocket_client.cpp

```diff
@@ -97,14 +99,20 @@ namespace signalr
             {
                 try
                 {
-
                     std::string msg;
                     auto response = task.get();
                     if (response.message_type() == web::websockets::client::websocket_message_type::binary_message)
                     {
-                        throw signalr_exception("binary isn't supported currently");
+                        concurrency::streams::producer_consumer_buffer<uint8_t> b;
+                        response.body().read_to_end(b).get();
+                        auto t = b.create_ostream().streambuf().in_avail();
+                        msg.resize(t, ' ');
+                        b.create_istream().streambuf().getn(reinterpret_cast<uint8_t*>(&msg[0]), t);
+                    }
+                    else
+                    {
+                        msg = response.extract_string().get();
                     }
-                    msg = response.extract_string().get();
 
                     callback(msg, nullptr);
```

Comment:
move callback into branches so we could avoid the original std::string initialization